include/readdy/common/Timer.h

```diff
@@ -34,76 +40,104 @@
 
 #include <memory>
 #include <chrono>
+#include <utility>
 
 #include "logging.h"
+#include "common.h"
 
 NAMESPACE_BEGIN(readdy)
 NAMESPACE_BEGIN(util)
 
+struct PerformanceData {
+    using time = readdy::scalar;
+    PerformanceData(time t, std::size_t c) : cumulativeTime(t), count(c) {}
+    mutable time cumulativeTime = 0.;
```

Comment:
Instead of exposing mutable members, expose const function to record a performance datum